addons/website_sale/models/sale_order.py

```diff
@@ -365,7 +365,7 @@ class SaleOrderLine(models.Model):
 
     name_short = fields.Char(compute="_compute_name_short")
 
-    linked_line_id = fields.Many2one('sale.order.line', string='Linked Order Line', domain="[('order_id', '!=', order_id)]", ondelete='cascade')
+    linked_line_id = fields.Many2one('sale.order.line', string='Linked Order Line', domain="[('order_id', '=', order_id)]", ondelete='cascade', copy=False)
```

Comment:
Hello @fmdl 

It seems wrong to revert on a stable release the logic existing since a long time
```py
domain="[('order_id', '!=', order_id)]"
```